### ISSUE TITLE
[html5] FileReference cancel event not dispatched

### ISSUE THREAD

**Issue by acarioni** [openfl/openfl#2621]
**Describe the bug**

FileReference.browse doesn’t dispatch the cancel event when one presses the cancel button.

Sometimes it throws the following error too:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'lastModified')
    at __inputControl.onchange
```

The offending code is in the method FileReferece.browse,  where `file.lastModified` is called:
```
browse: function(typeFilter) {
// ...
this.__inputControl.onchange = function() {
	var file = _gthis.__inputControl.files[0];
	var tmp = new Date(file.lastModified);
	_gthis.modificationDate = tmp;
	_gthis.creationDate = _gthis.get_modificationDate();
	_gthis.size = file.size;
	var tmp = haxe_io_Path.extension(file.name);
	_gthis.type = "." + tmp;
	_gthis.name = haxe_io_Path.withoutDirectory(file.name);
	_gthis.__path = file.name;
	_gthis.dispatchEvent(new openfl_events_Event("select"));
};
```

**Environment**

macOS 12.6.3
haxe 4.2.5
openfl 9.2.1
chrome 110.0.5481.177

**To Reproduce**

Create an openfl project like this
```
class Main extends Sprite {
  final file = new FileReference();
  public function new () {
    super ();
    file.addEventListener(Event.CANCEL, e -> trace(e));
    file.addEventListener(Event.SELECT, e -> trace(e));
    this.stage.addEventListener(MouseEvent.CLICK, e -> {
      trace(e);
      file.browse();
    });
  }
}
```

**Comment by joshtynjala** [comment]
OpenFL must use `<input type="file">` to open a browse dialog in the html5 target. Unfortunately, `<input type="file">` does not dispatch any kind of event when the cancel button is pressed, so there's no reliable way to detect it and dispatch our own event.

**Comment by acarioni** [comment]
Is it possible to add some notes on the docs of FileReference about the different behaviors among the targets?

**Comment by dimensionscape** [comment]
Sure, feel free to make a PR.

**Comment by acarioni** [comment]
I think this issue should be reopen because the exception thrown by the Cancel button was not fixed.

**Comment by joshtynjala** [comment]
In case it's not clear, the bug fix has not been released to Haxelib yet. If you want to use the fix, you need to install OpenFL's develop branch from Github for now.

**Comment by joshtynjala** [comment]
I just tested again to confirm. With OpenFL 9.2.1, I can reproduce the exception. With the develop branch from Github, I cannot.

I used the code above and built for the html5 target. I ran it in Chrome 111.0.5563.64. I clicked the stage once, and selected a file. Then, I clicked the stage again, and hit cancel. With OpenFL 9.2.1, I saw the exception about `lastModified`. With develop, I saw no exception, and `Event.CANCEL` was dispatched.
